### ISSUE TITLE
Error when trying to remotely execute JS in the browser

### ISSUE THREAD

**Issue by Darker** [saucelabs/foxdriver#14]
I am trying to use this library to remotely control netflix while building my own "smart tv". I am testing evaluating simple JS:
```
        const { browser, tab } = await Foxdriver.launch({
            url: 'https://www.netflix.com',
            bin: firefoxPath
        });
        await tab.console.startListeners();
        await new Promise((resolve) => setTimeout(resolve, 20000));
        console.log(await tab.console.evaluateJS("window.location.hash = \"blablabla\""));
```
The error:
```
(node:271348) UnhandledPromiseRejectionWarning: Error: this.parentActor.threadActor is undefined (unknownError)
    at D:\JAKUB\programming\node-smart-tv\SmartTV\node_modules\foxdriver\build\actor.js:61:23
    at Generator.next (<anonymous>)
    at step (D:\JAKUB\programming\node-smart-tv\SmartTV\node_modules\babel-runtime\helpers\asyncToGenerator.js:17:30)
    at D:\JAKUB\programming\node-smart-tv\SmartTV\node_modules\babel-runtime\helpers\asyncToGenerator.js:28:13
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

What's wrong? I need to evaluate commands to control the player.

**Comment by aberforthQ** [comment]
I can reproduce this too. I think it's an issue with Firefox: https://stackoverflow.com/questions/58054323/evaluating-code-remotely-in-firefox-from-node-js-this-parentactor-threadactor

**Comment by Darker** [comment]
I ended up writing my own library from scratch, that only supports the subset I need - executing script from console.

**Comment by aberforthQ** [comment]
Is the library open source? If so where can we find it? :-D

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by benmalka** [comment]
I think that synchronous script evaluation is no longer supported by Firefox API. I've implemented asynchronous evaluation in my fork (https://www.npmjs.com/package/@benmalka/foxdriver). If you have the same issue there, please open an issue the fork repo

**Comment by benmalka** [comment]
Should be fixed in #22. evaluateJS is no longer supported by Firefox, only async one

**Comment by christian-bromann** [comment]
Closing then, thanks @benmalka
